x-pack/plugin/inference/qa/test-service-plugin/src/main/java/org/elasticsearch/xpack/inference/mock/TestDenseInferenceServiceExtension.java

```diff
@@ -215,6 +215,12 @@ public record TestServiceSettings(
 
         static final String NAME = "test_text_embedding_service_settings";
 
+        public TestServiceSettings {
+            if (elementType == DenseVectorFieldMapper.ElementType.BIT) {
+                throw new IllegalArgumentException("Test dense inference service does not yet support element type BIT");
+            }
+        }
```

Comment:
I have a plan for how to update `TestDenseInferenceServiceExtension` extension, which I will handle in a follow-up PR.